### ISSUE TITLE
Add FAQ or Example about multi-step forms

### ISSUE THREAD

**Issue by dandv** [vazco/uniforms#730]
I'm evaluating React form libraries and have two key criteria:

1. Support for multi-step/wizard forms (they show [much](https://growrevenue.io/why-multi-step-lead-forms-get-more-conversions/) better [conversion](https://www.ventureharbour.com/multi-step-lead-forms-get-300-conversions/))
2. Autogenerating the form based on GraphQL schema or TypeScript classes

RHF [supports multi-step forms](https://codesandbox.io/s/form-wizard-pages-kkg7m), but [autogeneration is not on the roadmap](https://spectrum.chat/react-hook-form/help/build-form-from-class-mongo-schema-definition~a36a612d-f358-4f69-9198-59901f695eda).

I couldn't find any documentation about multi-step forms at https://uniforms.tools/docs/ but I do see #383 and #719 here about that topic, so apparently it's possible?

For new users evaluating form libraries, making the multi-step capability easily visible would be important in helping them choose uniforms.

**Comment by hmvp** [comment]
We currently implement multi step forms with uniforms, but the multi step part is managed by a component that uses uniforms for each step and the individual pages/uniform instances don't need to interface with each other. 

I did not find any issues with it, but our way of handling forms might not be very typical. We send each page on submit to the backend for validation and to determine the next page..

**Comment by radekmie** [comment]
Hi @dandv, @hmvp! This topic comes back from time to time, yet it's not covered in the docs. It should be, we know that. The thing is that it's often not the actual thing one is asking for.

_(I'll elaborate here on this topic and hopefully will use this thread as a base for the doc page)_

What _is_ a multi-step form? Well, I can imagine at least two completely separate definitions:
1. **A set of independent forms with a shared state.** That's the _easier_ one as it's always possible. That's exactly what [this `react-hook-form` demo](https://codesandbox.io/s/form-wizard-pages-kkg7m) does, and @hmvp explains in https://github.com/vazco/uniforms/issues/730#issuecomment-631270897. Each step renders a separate form, with a different schema/validator/style and moves to the next one when submitted.
 
    This handles not only multi-step forms but also ~forms~ wizards with a tree-like structure (i.e. next step bases on the answers). Optional steps (_skip step 2 if age < 40_) and contextual validation (_field Y in step 2 has to be greater than the value of X in step 1_) is also possible.

    But it gets even better - each step may use a different forms library! It makes no sense but is definitely possible - each form is independent, and the orchestration happens in the application.
2. **A single form displayed in parts.** It is, of course, possible to implement it, but the number of all configurations and options is _massive_. But let's skip that and see where a bigger problem is: the validation. In **1.** each step is validated separately (i.e. can have a separate schema). Here, we have only one schema, and the schema itself has to know that _some_ fields were not yet visible.

    Let's make an example. Our schema is very basic: `{ a: string, b: string }` (TypeScript notation). Now, as both `a` and `b` are required, a _valid_ model have to have both. If the first step will render only the `a` field (`b` is on the next page), it's impossible to validate the form. This leads to a situation where the schema (logic) depends on the steps (UI). On the other hand, the form could be validated only at the end. The UX of this solution is terrible though - imagine a _there's an error ten pages back_ error!

**TL;DR;** We are not planning to provide any out-of-the-box support for multi-step forms as option **1.** is most of the time the best. It's not only the cleanest but also less complicated as well as doesn't rely on any library.

**Comment by ymoussaba** [comment]
The simplest and quickest way to do this in my opinion is to use a single schema and divide the fields into different slides with a carousel as follows.
You can find a demo here : https://codesandbox.io/s/jlgft
